python/fusion_engine_client/utils/enum_utils.py

```diff
@@ -12,8 +12,28 @@ def __str__(self):
         #   print(Foo.BAR)   # Prints "BAR", not "Foo.BAR"
         return self.name
 
-    def to_string(self, include_value=False):
+    def to_string(self, include_value=True):
         if include_value:
             return '%s (%d)' % (str(self), int(self))
         else:
             return str(self)
+
+    @classmethod
+    def static_to_string(cls, value, include_value=True, raise_on_unrecognized=False):
```

Comment:
From my previous comments I guess I see how this works now, but I still don't understand the need for the explicitly called classmethod or the call to super. It seems like this would be simpler to just do as __repr__ or as instance methods.
I guess for the main use case it seems simpler to call:
`MessageType(value).display_string()`
or
`repr(MessageType(value))`
depending on how you feel about __repre__.

Reply:
So we do have `str()` support as a class member function (not `repr()`, we just let that do its own thing), and this function ultimately uses that in the `str(self)` calls to get the enum value's name. The point of having this function in addition is that we can gracefully support integer values that are not defined as enums in the version of the code you're working with.

For example, say we add a new `EventType::STUFF = 6` in the C++ on the device, but you're using an older version of the Python code. The previous code would do something like:
```py
# Unpack:
self.event_type = EventType(int_value)

# Later, expecting:
#   Event: STUFF (6)
print('Event: %s' % str(self.event_type))
```
but since `STUFF` isn't defined, that `EventType()` call would actually raise an exception, and the message would fail to parse. Or even if the parsing function is modified to handle the unknown value and set `self.event_type = 6`, the user code might raise an exception trying to print the value out.

Using this function, we will now display `Event: <Unrecognized> (6)` without crashing. That way we can add new definitions without having things hit exceptions unexpectedly in cases where they could do something more graceful. Of course, if you need to do something specific with `EventType.STUFF`, you will need to update your Python requirements to do that regardless.